### PR TITLE
Render a banner and version info on server startup

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: server/src/main/scala/org/http4s/server/Server.scala [hunk 1]
```diff
@@ -16,6 +19,25 @@ abstract class Server[F[_]](implicit F: Effect[F]) {
 
   def address: InetSocketAddress
 
+  def isSecure: Boolean
+
+  def baseUri: Uri = Uri(
```
Comment (rossabaker) [inline server/src/main/scala/org/http4s/server/Server.scala]: This is why our URI type needs reform. ☣️

Comment (ChristopherDavenport) [reply server/src/main/scala/org/http4s/server/Server.scala]: Ugh.

File: server/src/main/scala/org/http4s/server/ServerBuilder.scala [hunk 2]
```diff
@@ -62,6 +69,12 @@ object ServerBuilder {
   val DefaultHost = LoopbackAddress
   val DefaultHttpPort = 8080
   val DefaultSocketAddress = InetSocketAddress.createUnresolved(DefaultHost, DefaultHttpPort)
+  val DefaultBanner =
```
Comment (ChristopherDavenport) [inline server/src/main/scala/org/http4s/server/ServerBuilder.scala]: Love it

**Comment by rossabaker** [comment]
This will have some subtle conflicts in the merge train.

**Pull request merged**